flowmachine/flowmachine/core/server/action_handlers.py

```
@@ -80,6 +85,16 @@ async def action_handler__get_query_schemas(
     try:
         query_obj = FlowmachineQuerySchema().load(params)
+        query_obj._flowmachine_query_obj.preflight()  # Note that we probably want to remove this call to allow getting qid faster
+    except PreFlightFailedException as exc:
+        orig_error_msg = exc.args[0]
+        error_msg = (
+            f"Internal flowmachine server error: could not create query object using query schema. "
+            f"The original error was: '{orig_error_msg}'"
+        )
+        raise QueryLoadError(
+            error_msg, params, orig_error_msg=orig_error_msg, errors=exc.errors
+        )
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)